crates/integration_tests/src/lib.rs

```diff
@@ -18,31 +18,32 @@
 use std::collections::HashMap;
 
 use iceberg::io::{S3_ACCESS_KEY_ID, S3_ENDPOINT, S3_REGION, S3_SECRET_ACCESS_KEY};
-use iceberg_catalog_rest::{RestCatalog, RestCatalogConfig};
+use iceberg_catalog_rest::RestCatalogConfig;
 use iceberg_test_utils::docker::DockerCompose;
 use iceberg_test_utils::{normalize_test_name, set_up};
 
 const REST_CATALOG_PORT: u16 = 8181;
 
 pub struct TestFixture {
     pub _docker_compose: DockerCompose,
-    pub rest_catalog: RestCatalog,
+    pub catalog_config: RestCatalogConfig,
```

Comment:
Since the tests now share this fixture but they're all executed in different runtimes, they can't share the same client, as those can get [dropped prematurely](https://github.com/seanmonstar/reqwest/issues/1148) when a test ends, resulting in `dispatch task is gone: runtime dropped the dispatch task`